### PR TITLE
XW-1304 | Separate Map from ImageMedia component

### PR DESCRIPTION
**Pull request by vforge**
## Links

* http://wikia-inc.atlassian.net/browse/XW-1304
* https://github.com/Wikia/app/pull/10137

## Description

Separate Map from ImageMedia component

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: front/main/app/components/article-content.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import GalleryMediaComponent from './gallery-media';
 import VideoMediaComponent from './video-media';
 import ImageMediaComponent from './image-media';
 import InfoboxImageCollectionComponent from './infobox-image-collection';
-import WikiaMapComponent from './wikia-map';
+import WikiaMapComponent from './article-media-map-thumbnail';
```
Comment (rogatty) [inline front/main/app/components/article-content.js]: It's not used.

File: front/main/app/components/article-wrapper.js [hunk 2]
```diff
@@ -223,10 +223,19 @@ export default Ember.Component.extend(
 		 * @returns {boolean}
 		 */
 		click(event) {
-			const $anchor = Ember.$(event.target).closest('a');
+			const $map = Ember.$(event.target).closest('.article-media-map-thumbnail'),
+				$anchor = Ember.$(event.target).closest('a');
 
 			let target;
 
+			// Wikia Map - it's not a link, but we handled it already
```
Comment (rogatty) [inline front/main/app/components/article-wrapper.js]: Let's prevent the situation where click on map bubbles here.

File: front/main/app/styles/component/_article-media.scss [hunk 3]
```diff
@@ -50,6 +50,38 @@
 	}
 }
 
+.article-media-map-thumbnail {
+  margin: 40px 0;
```
Comment (rogatty) [inline front/main/app/styles/component/_article-media.scss]: Use tabs.

File: front/main/app/styles/component/_article-media-map-thumbnail.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  a {
+	display: block;
+  }
+
+  .icon {
+	fill: #fff;
+	position: absolute;
+	top: 5%;
+	right: 5%;
```
Comment (bognix) [inline front/main/app/styles/component/_article-media-map-thumbnail.scss]: alphabetical ordering

**Comment by bognix** [comment]
LGTM - minor notes

File: front/main/app/components/article-content.js [hunk 5]
```diff
@@ -247,6 +245,12 @@ export default Ember.Component.extend(
 				}
 			}
 
+			if (name === 'article-media-map-thumbnail') {
```
Comment (rogatty) [inline front/main/app/components/article-content.js]: `else if`?

File: front/main/app/styles/component/_article-media-gallery.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
+.article-media-gallery {
+  -webkit-overflow-scrolling: touch;
```
Comment (rogatty) [inline front/main/app/styles/component/_article-media-gallery.scss]: Use tabs in all SCSS files.

File: front/main/app/styles/component/_article-media-map-thumbnail.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	padding: 5px 0;
+	position: absolute;
+	text-overflow: ellipsis;
+	white-space: nowrap;
+	width: 90%;
+  }
+}
+
+
+
```
Comment (rogatty) [inline front/main/app/styles/component/_article-media-map-thumbnail.scss]: It's too much! :see_no_evil:

File: front/main/app/styles/component/_article-media-map-thumbnail.scss [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  .icon {
+	fill: #fff;
+	position: absolute;
+	top: 5%;
+	right: 5%;
+  }
+
+  figcaption {
```
Comment (rogatty) [inline front/main/app/styles/component/_article-media-map-thumbnail.scss]: Shouldn't elements be above the classes or the other way around? Right now it's:
```
a {}
.icon {}
figcaption {}
```

File: front/main/app/styles/component/_article-media-map-thumbnail.scss [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+	bottom: 0;
+	color: #fff;
+	font-size: 1.6rem;
+	font-weight: 700;
+	height: 50px;
+	left: 5%;
+	overflow: hidden;
+	padding: 5px 0;
+	position: absolute;
+	text-overflow: ellipsis;
```
Comment (rogatty) [inline front/main/app/styles/component/_article-media-map-thumbnail.scss]: `@include ellipsis`?

File: front/main/app/templates/components/article-media-map-thumbnail.hbs [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<a href="{{url}}">
+	<img src="{{imageSrc}}">
+	{{svg 'view-map' viewBox='0 0 20 20' class='icon'}}
+</a>
+{{#if title}}
+    <figcaption>{{{title}}}</figcaption>
```
Comment (rogatty) [inline front/main/app/templates/components/article-media-map-thumbnail.hbs]: Use `\t`

**Comment by rogatty** [comment]
LGTM

**Pull request merged**